drivers/middleware/gin/options.go

```diff
@@ -62,3 +62,25 @@ func WithKeyGetter(KeyGetter KeyGetter) Option {
 func DefaultKeyGetter(c *gin.Context) string {
 	return c.ClientIP()
 }
+
+// ExcludedKey is function type used to check whether the key should be excluded or not
+type ExcludedKey func(key string) bool
+
+// DefaultExcludedKey is the default function returns ExcludedKey
+func DefaultExcludedKey(keys []string) ExcludedKey {
+	m := make(map[string]struct{}, len(keys))
+	for _, key := range keys {
+		m[key] = struct{}{}
+	}
+	return func(key string) bool {
+		_, ok := m[key]
+		return ok
+	}
+}
+
+// WithExcludedKey will configure the Middleware to use the given ExcludedKey.
+func WithExcludedKey(fn ExcludedKey) Option {
```

Comment:
I don't understand why you create a configurable exclusion function. Do you have an internal need to change the `DefaultExcludedKey` behavior ?

I think it would be better to change this option signature to `func WithExcludedKey(keys []string) Option`. What do you think ?

Reply:
`ExcludedKey`  has more flexibility, for example,  user can define a function which to handle non-enumerable keys:

```
prefix := "prefix"
func ExcludedKey(key string) bool {
    return strings.HasPrefix(key, prefix)
}
```

Reply:
@novln PTAL

Reply:
That's a fair point, however I feel like having two functions for this is a bit cumbersome:

```
gin.WithExcludedKey(gin.DefaultExcludedKey([]string{"1"}))
```

What you could do, and in my opinion is the best trade-off, would be to just expose `WithExcludedKey` function _(removing the DefaultExcludedKey function and even `ExcludedKey`, but I don't have strong opinion for this last one)_, leaving to the end-user what implementation they would want _(like accessing from database, configuration files or static declaration)_.

This way we keep the API minimal and we ensure flexibility at the same time.